subsys/usb/os_desc.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_USB_DEVICE_LEVEL
+#define SYS_LOG_DOMAIN "usb/osdesc"
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+
+#include <usb/usb_device.h>
+#include <os_desc.h>
+
+static struct usb_os_descriptor *os_desc;
+
+int usb_handle_os_desc(struct usb_setup_packet *setup,
+		       s32_t *len, u8_t **data)
+{
+	SYS_LOG_DBG("wValue 0x%x", setup->wValue);
+
+	if (!os_desc) {
+		return -ENOTSUP;
+	}
+
+	if (GET_DESC_TYPE(setup->wValue) == DESC_STRING &&
+	    GET_DESC_INDEX(setup->wValue) == 0xEE) {
```

Comment:
Can 0xEE be replaced by a macro or otherwise documented?

Reply:
This is possible, I will add documentation block.